app/src/organisms/Devices/PipetteCard/PipetteOverflowMenu.tsx

```diff
@@ -51,49 +59,55 @@ export const PipetteOverflowMenu = (
         right={`calc(50% + ${SPACING.spacing2})`}
         flexDirection={DIRECTION_COLUMN}
       >
-        {pipetteName === 'Empty' ? (
+        {pipetteDisplayName === 'Empty' ? (
           <MenuItem
-            key={`${pipetteName}_${mount}_attach_pipette`}
+            key={`${pipetteDisplayName}_${mount}_attach_pipette`}
             onClick={() => handleChangePipette()}
-            data-testid={`pipetteOverflowMenu_attach_pipette_btn_${pipetteName}_${mount}`}
+            data-testid={`pipetteOverflowMenu_attach_pipette_btn_${pipetteDisplayName}_${mount}`}
           >
             {t('attach_pipette')}
           </MenuItem>
         ) : (
           <>
+            {!isOT3PipetteAttached && (
+              <MenuItem
+                key={`${pipetteDisplayName}_${mount}_calibrate_offset`}
+                onClick={() => handleCalibrate()}
+                data-testid={`pipetteOverflowMenu_calibrate_offset_btn_${pipetteDisplayName}_${mount}`}
+              >
+                {t(
+                  isPipetteCalibrated
+                    ? 'recalibrate_pipette_offset'
+                    : 'calibrate_pipette_offset'
+                )}
+              </MenuItem>
+            )}
+            {!isOT3PipetteAttached && (
+              <MenuItem
+                key={`${pipetteDisplayName}_${mount}_detach`}
+                onClick={() => handleChangePipette()}
+                data-testid={`pipetteOverflowMenu_detach_pipette_btn_${pipetteDisplayName}_${mount}`}
+              >
+                {t('detach_pipette')}
+              </MenuItem>
+            )}
```

Comment:
This logic can be simplified

```suggestion
             {!isOT3PipetteAttached && (
              <>
                <MenuItem
                  key={`${pipetteDisplayName}_${mount}_calibrate_offset`}
                  onClick={() => handleCalibrate()}
                  data-testid={`pipetteOverflowMenu_calibrate_offset_btn_${pipetteDisplayName}_${mount}`}
                >
                  {t(
                    isPipetteCalibrated
                      ? 'recalibrate_pipette_offset'
                      : 'calibrate_pipette_offset'
                  )}
                </MenuItem>

                <MenuItem
                  key={`${pipetteDisplayName}_${mount}_detach`}
                  onClick={() => handleChangePipette()}
                  data-testid={`pipetteOverflowMenu_detach_pipette_btn_${pipetteDisplayName}_${mount}`}
                >
                  {t('detach_pipette')}
                </MenuItem>
              </>
            )}
```